charts/atlantis/README.md

```diff
@@ -229,6 +230,86 @@ extraManifests:
 
 ## Upgrading
 
+### From `4.*` to `5.*`
+
+A breaking change was merged on [#304](https://github.com/runatlantis/helm-charts/pull/304).
+This change moves the atlantis data volume to a separate PVC to allow changing the volume size without removing the statefulset.
+Unmistakingly, this change only bumped the minor version, released as `4.22.0`, instead of incrementing the major version. The release `5.0.0` addresses this issue. Please find the required upgrade manual steps below.
+
+#### Upgrade steps
+
+Trying to upgrade the release directly will return the following error:
+
+> Error: UPGRADE FAILED: cannot patch "atlantis" with kind StatefulSet: StatefulSet.apps "atlantis" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'ordinals', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden
+
+To allow the upgrade to proceed, first we need to remove the statefulset controller, copy all the data from the old volume to the new one and then cleanup.
+
+NOTE: These steps will ensure no data is lost, however during the copy process the atlantis server is unavailable and any webhooks sent to atlantis during that window are missed and won't be processed.
+
+Upgrade steps:
+
+1. For atlantis deployments managed by GitOps solutions such as ArgoCD, disabling the automatic synchronization is required
+
+1. Create a temporary helm configuration file named `migration-4-to-5.yaml`, the configuration includes the required settings to allow the atlantis data to be copied to the new volume before the atlantis server is started. The suggested configuration implies that the default command for atlantis has not been customized. If required, adjust the `persistentVolumeClaim.claimName` value.
+
+    ```yaml
+    command:
+      - sh
+      - -c
+      - 'cp -Rv /atlantis-data-old/* "${ATLANTIS_DATA_DIR}" && exec /usr/local/bin/docker-entrypoint.sh server'
+
+    extraVolumes:
+      - name: atlantis-data-old
+        persistentVolumeClaim:
+          claimName: atlantis-data-atlantis-0
+
+    extraVolumeMounts:
+      - name: atlantis-data-old
+        mountPath: /atlantis-data-old
```

Comment:
Shouldn’t the names of the above PV & PVC be the same as pre chart v4.22.0? i.e. by old default `atlantis-data`? 

will I need to define a different value for `ATLANTIS_DATA_DIR` to avoid directory name conflict?

Reply:
The PVC name needs to match whatever is there before 4.22. Don’t you have the same as the examples? Can you share pls?

Note that the mount is just to copy data. You will then remove them.

Reply:
I just re-confirmed that the claim name is correct:

```sh
→ helm install atlantis runatlantis/atlantis --version 4.21.1
NAME: atlantis
LAST DEPLOYED: Sat May  4 12:10:14 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
1. Get the application URL by running these commands:
2. Atlantis will not start successfully unless at least one of the following sets of credentials are specified (see values.yaml for detailed usage):
  - github
  - githubApp
  - gitlab
  - bitbucket
```
```sh
→ kubectl get pvc
NAME                       STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
atlantis-data-atlantis-0   Pending                                      standard       <unset>                 4s
```